Add support for configuring the border radius using theme var

giuliandenicola1 opened the pull request:
Hi, I've made some visual changes in the CSS. 
See if you like it. 

I've added another theme option name: sonos-border-radius

<img width="1440" alt="Schermafbeelding 2022-02-25 om 11 53 29" src="https://user-images.githubusercontent.com/67443916/155707262-4ce1c3a1-8fb5-4533-84ba-a66ab86bc5e3.png">
<img width="1440" alt="Schermafbeelding 2022-02-25 om 11 54 53" src="https://user-images.githubusercontent.com/67443916/155707324-96510797-5903-4d14-8219-2b24d744be08.png">

cheers.

johanfrick commented on src/main.ts:
```diff
@@ -216,7 +216,7 @@ export class CustomSonosCard extends LitElement {
       .players {
         flex: 0 0 45%;
-        max-width: 45%;
+        max-width: 700px;
```
I'd like to keep this as is. There is an upcoming change that makes this configurable: https://github.com/johanfrick/custom-sonos-card/commit/12f8ed3b2434380f3e86db6ece531e32338ef000

giuliandenicola1 replied:
I see, this will be a great feature to have! No need for the max width to be implemented if it's going to be like that.

johanfrick replied:
:-) So until that is out, can you revert this line?

giuliandenicola1 replied:
Ow! Sure! Didn't know you meant that.

johanfrick replied:
Layout is now configurable with https://github.com/johanfrick/custom-sonos-card/releases/tag/v2.4.0

https://github.com/johanfrick/custom-sonos-card#layout

johanfrick commented on src/main.ts:
```diff
@@ -237,7 +237,7 @@ export class CustomSonosCard extends LitElement {
       }
       .title {
-        margin-top: 10px;
+        margin-top: 10px 0;
```
What does the extra zero do?

giuliandenicola1 replied:
I see that I've made a mistake, it needs to be `margin: 10px 0;` so in theory that makes it `margin: 10px 0 10px 0;`

johanfrick commented on src/player.ts:
```diff
@@ -146,7 +146,6 @@ class Player extends LitElement {
         position: relative;
         background: var(--sonos-int-background-color);
         border-radius: var(--sonos-int-border-radius);
-        border: 8px solid var(--sonos-int-background-color);
```
A border here is probably nice when having a background image in your lovelace theme. Could we make it themable?

giuliandenicola1 replied:
You're right! I didn't think of that. It could be like this `border: var(--sonos-player-border) solid var(--sonos-int-background-color);`

johanfrick commented on src/favorite-buttons.ts:
```diff
@@ -58,6 +58,7 @@ class FavoriteButtons extends LitElement {
         flex-wrap: wrap;
       }
       .favorite {
+        overflow: hidden;
```
What problem did you see when not having this?

giuliandenicola1 replied:
The background behind the text is a square so if you use a border radius then you would see the corners popping out of the card. Might be better to add a border radius to the favorites title card when I think about it now.

johanfrick commented:
Thanks for contributing! I would hope we can make it all configurable so the user can decide if they want (very) rounded corners or not :-)

giuliandenicola1 commented:
I've made changes based on your reply, hope this is what you talked about. 
Added the `sonos-border-width:` option for the theme file.

I've removed the overflow hidden now the favorites looks like this 
<img width="228" alt="Schermafbeelding 2022-02-25 om 14 04 18" src="https://user-images.githubusercontent.com/67443916/155720083-852b1e9e-c997-4967-a108-72d7a62eaafa.png">

Instead of this
<img width="289" alt="Schermafbeelding 2022-02-25 om 14 04 32" src="https://user-images.githubusercontent.com/67443916/155720109-34cd1911-935a-4089-9845-519da8416b91.png">

johanfrick commented:
Nice! There is one more question above, please check :-)

giuliandenicola1 commented:
> Nice! There is one more question above, please check :-)

Awnsered!

johanfrick commented on src/main.ts:
```diff
@@ -198,7 +198,8 @@ export class CustomSonosCard extends LitElement {
         --sonos-int-title-color: var(--sonos-title-color, var(--card-background-color));
-        --sonos-int-border-radius: 0.25rem;
+        --sonos-int-border-radius: var(--sonos-border-radius, 0.25rem);
+        --sonos-int-border-width: var(--sonos-border-width, 8px);
```
I'd like to use rem unit, is that possible?

giuliandenicola1 replied:
Sure, I only used the 8px because it was there already. I only use rem where you used rem and only px where you used px. Be free to change anything I do I don't mind.

johanfrick commented:
Again, nice addition! Could you update the readme with the new variable?

giuliandenicola1 commented:
I've updated the readme and changed the border width from px to rem

johanfrick reviewed (approved):
💪

johanfrick commented:
@giuliandenicola1 https://github.com/johanfrick/custom-sonos-card/releases/tag/v2.3.5
